src/protocol/extension-files.ts

```diff
@@ -0,0 +1,28 @@
+import { TextDocumentIdentifier, TextDocumentItem } from 'vscode-languageserver-types'
+import { RequestHandler } from '../jsonrpc2/handlers'
+import { RequestType } from '../jsonrpc2/messages'
+import { TextDocumentRegistrationOptions } from './textDocument'
+
+export interface FilesExtensionClientCapabilities {
```

Comment:
add docs to most types in this file; this file's docstrings can serve as the protocol spec, and then there's no need to rely on the LSP extension draft you linked elsewhere